lightly_studio/src/lightly_studio/plugins/parameter.py

```diff
@@ -0,0 +1,132 @@
+"""Parameter for operators for LightlyStudio plugins."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class BaseParameter(ABC):
+    """Base parameter definition shared across operator parameters."""
+
+    __slots__ = ("default", "description", "name", "param_type", "required")
+
+    @abstractmethod
+    def __init__(
+        self,
+        *,
+        name: str,
+        description: str = "",
+        default: Any = None,
+        required: bool = True,
+    ) -> None:
+        """Initialize the parameter."""
+        self.name = name
+        self.description = description
+        self.required = required
+        self.default = self._validate(default) if default is not None else None
+
+    def to_dict(self) -> dict[str, Any]:
+        """Convert parameter to dictionary representation."""
+        return {
+            "name": self.name,
+            "type": self.param_type,
+            "description": self.description,
+            "default": self.default,
+            "required": self.required,
+        }
+
+    @staticmethod
+    @abstractmethod
+    def _validate(value: Any) -> Any:
+        """Validate the parameter value."""
+
+
+class IntParameter(BaseParameter):
```

Comment:
Did you consider having a generic for these types? This would save us a lot of code to review and tests.

It would change the logic such that a bool is then a valid value for a float or int parameter, but that is fine for me.